daemon/algod/api/server/v2/handlers.go

```diff
@@ -1114,6 +1115,64 @@ func (v2 *Handlers) GetApplicationByID(ctx echo.Context, applicationID uint64) e
 	return ctx.JSON(http.StatusOK, response)
 }
 
+// GetApplicationBoxes returns the box names of an application
+// (GET /v2/applications/{application-id}/boxes)
+func (v2 *Handlers) GetApplicationBoxes(ctx echo.Context, applicationID uint64, params generated.GetApplicationBoxesParams) error {
+	appIdx := basics.AppIndex(applicationID)
+	ledger := v2.Node.LedgerForAPI()
+	lastRound := ledger.Latest()
+	keyPrefix := logic.MakeBoxKey(appIdx, "")
+
+	// cast param.Max to zero, if param.Max points to some value, we keep the value
+	castedMax := nilToZero(params.Max)
+	maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication
+
+	// compute `maxReturnNum` from `castedMax` and `maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication`
+	// `maxReturnNum == 0` for returning everything
+	// - `castedMax == 0` and `maxBoxThreshold == 0`,
+	//       return everything
+	// - `castedMax == 0` and `maxBoxThreshold > 0`,
+	//       returns up to `maxBoxThreshold`, check if exceeds
+	// - `castedMax > 0` and `maxBoxThreshold == 0`,
+	//       return up to `castedMax`
+	// - `castedMax > 0` and `maxBoxThreshold > 0`,
+	//       return up to min(castedMax, maxBoxThreshold),
+	//       if return size is `maxBoxThreshold`, check if exceeds
+
+	var boxKeys []string
+	var err error
+
+	dominatedByQryParams := castedMax > 0 && (maxBoxThreshold > castedMax || maxBoxThreshold == 0)
```

Comment:
```suggestion
	dominatedByQryParams := castedMax > 0 && (maxBoxThreshold >= castedMax || maxBoxThreshold == 0)
```
I think we want `>=`. Because we want to be in the first clause where you do _not_ error if there are more values, when the supplied max is equal to the configuration max.  The caller is agreeing to receive that maximum.